Players can join and play as a template-less PlayerSpaceship

In a scenario file, add a `PlayerSpaceship()`, but don't add a `setTemplate("...")` function.

OR

With a running scenario, use the HTTP API to execute `PlayerSpaceship()` without a template.

**Expected behavior**

A script error prevents the template-less PlayerSpaceship from being created, because a template is required.

OR

A template-less PlayerSpaceship can be created, but it does not show up on the list of joinable ships to clients, and cannot be joined by players, until it has a template applied to it.

**Observed behavior**

A ship with no template is created and listed as a class-less Human Navy player ship. It can be joined by the Helms, Science, and Relay stations on all clients, including the client running as the server. Helms can pilot the ship with impulse, warp, and jump drives, Science can scan enemy ships, and Relay can launch probes and hail stations and ships.

If a player joins the Weapons or Engineering stations on the server's client, the server crashes on a segfault. Clients not running as the server can join the Weapons station, but crash when attempting to join the Engineering station.

The templateless ship is visible on the GM screen as a small purple dot, with a generated `PL`-prefixed callsign. The GM can select the ship and move it. Attempting to tweak it crashes the server.

While any templateless ship is active, the console logs flood with a line that repeats every tick:

```
[ERROR] Failed to find ship template:
```

<img width="582" alt="Screen Shot 2020-12-28 at 7 20 59 PM" src="https://user-images.githubusercontent.com/19192104/103256787-56a19a80-4943-11eb-9fe1-015b1c477c07.png">
<img width="1192" alt="Screen Shot 2020-12-28 at 7 21 21 PM" src="https://user-images.githubusercontent.com/19192104/103256790-57d2c780-4943-11eb-8f89-4e747b2fa91d.png">
<img width="1188" alt="Screen Shot 2020-12-28 at 7 26 51 PM" src="https://user-images.githubusercontent.com/19192104/103256795-599c8b00-4943-11eb-9a2c-cf7ed3a4c279.png">

--- a/src/spaceObjects/shipTemplateBasedObject.cpp
+++ b/src/spaceObjects/shipTemplateBasedObject.cpp
@@ -219,11 +219,22 @@ void ShipTemplateBasedObject::draw3DTransparent()
 
 void ShipTemplateBasedObject::update(float delta)
 {
+    // All ShipTemplateBasedObjects should have a valid template.
+    // If this object lacks a template, or has an inconsistent template...
     if (!ship_template || ship_template->getName() != template_name)
     {
+        // Attempt to align the object's template to its reported template name.
         ship_template = ShipTemplate::getTemplate(template_name);
+
+        // If the template still doesn't exist, destroy the object.
         if (!ship_template)
+        {
+            LOG(ERROR) << "ShipTemplateBasedObject with ID " << string(getMultiplayerId()) << " lacked a template, so it was destroyed.";
+            destroy();
             return;
+        }
+
+        // If it does exist, set up its collider and model.
         ship_template->setCollisionData(this);
         model_info.setData(ship_template->model_data);
     }

